als-server/shared/src/main/scala/org/mulesoft/als/server/workspace/WorkspaceManager.scala

```diff
@@ -35,13 +36,21 @@ class WorkspaceManager protected (val environmentProvider: EnvironmentProvider,
 
   implicit val platform: Platform = environmentProvider.platform
 
-  override def subscribers: List[BaseUnitListener] = allSubscribers.filter(_.isActive)
+  override def subscribers: List[BaseUnitListener] =
+    allSubscribers.collect({ case buL: BaseUnitListener if buL.isActive => buL })
+
+  private def workspaceSubscribers: List[AstListener[_] with WorkspaceContentListener] =
+    allSubscribers.collect({
+      case ncL: NewConfigurationListener if ncL.isActive => ncL
```

Comment:
if `isActive` is part of the WorkspaceContentListener trait (or AstListener), this collect can avoid the hard match and instead just be a `filter`